packages/create-app/src/commands/init.ts

```diff
@@ -83,7 +83,7 @@ export default class Init extends Command {
         'Only GitHub repository references are supported, ' +
           'e.g., https://github.com/Shopify/<repository>/[subpath]#[branch]',
       )
-    if (!url && !allTemplates.includes(template))
+    if (!url && !isPredefinedTemplate(template))
       throw new AbortError(
         outputContent`Only ${visibleTemplates
           .map((alias) => outputContent`${outputToken.yellow(alias)}`.value)
```

Comment:
Should we also add a new flag to select the `branch`? but called flavor like for extensions
`create-app --template remix --flavor typescript`

Reply:
The way I set this up you'd pass in the template and get prompted for the branch, for `remix`. We could add a flag, but I (slightly) prefer the current experience.

Happy to go with whatever the team prefers.

Reply:
There is a "rule" that all prompts should have an alternative without user interaction (for nonTTY terminals for instance), but that's already covered by the `--template` flag if you use a full repo name like `https://github.com/Shopify/<repository>/[subpath]#[branch]` 🤔 
So not sure if it's worth it to add a new flag or not, probably a @MeredithCastile decission